server/src/main/java/org/opensearch/gateway/BaseGatewayShardAllocator.java

```diff
@@ -74,7 +78,41 @@ public void allocateUnassigned(
         ExistingShardsAllocator.UnassignedAllocationHandler unassignedAllocationHandler
     ) {
         final AllocateUnassignedDecision allocateUnassignedDecision = makeAllocationDecision(shardRouting, allocation, logger);
+        executeDecision(shardRouting, allocateUnassignedDecision, allocation, unassignedAllocationHandler);
+    }
+
+    /**
+     * Allocate Set of unassigned shard  to nodes where valid copies of the shard already exists
+     * @param shards the shards to allocate
+     * @param allocation the allocation state container object
+     */
+    public void allocateUnassignedBatch(Set<ShardRouting> shards, RoutingAllocation allocation) {
+        // make Allocation Decisions for all shards
+        HashMap<ShardRouting, AllocateUnassignedDecision> decisionMap = makeAllocationDecision(shards, allocation, logger);
+        assert shards.size() == decisionMap.size() : "make allocation decision didn't return allocation decision for " + "some shards";
+        // get all unassigned shards iterator
+        RoutingNodes.UnassignedShards.UnassignedIterator iterator = allocation.routingNodes().unassigned().iterator();
+
+        while (iterator.hasNext()) {
+            ShardRouting shard = iterator.next();
+            try {
+                if (decisionMap.isEmpty() == false) {
+                    if (decisionMap.containsKey(shard)) {
+                        executeDecision(shard, decisionMap.remove(shard), allocation, iterator);
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("Failed to execute decision for shard {} ", shard, e);
+            }
```

Comment:
When will this Exception be thrown ? Is this intentional `catch-and-log` for the shard; If yes, can we add a more specific Exception Type then generic `Exception` here.

Reply:
While executing decision.

Reply:
Other PRs are merged without ParameterizedMessage logging, it wasnt failing.

We are catching a generic exception here which is unforseen to swallow any exception while assiging shard so that other shards in batches gets there turn for assignment

Reply:
Modified error message.